applications/ShapeOptimizationApplication/python_scripts/algorithms/algorithm_relaxed_gradient_projection.py

```diff
@@ -0,0 +1,548 @@
+
+        settings = KM.Parameters('{ "solver_type" : "LinearSolversApplication.dense_col_piv_householder_qr" }')
+        solver = dense_linear_solver_factory.ConstructSolver(settings)
+
+        c = KM.Vector()
+
+        KM.Logger.PrintInfo("ShapeOpt", "Calculate projected search direction and correction.")
+        self.optimization_utilities.CalculateRelaxedProjectedSearchDirectionAndCorrection(
+            nabla_f,
+            N,
+            omega_r,
+            omega_c,
+            solver,
+            p,
+            c)
+
+        # additional normalization step
+
+        self.optimization_utilities.AssignVectorToVariable(self.design_surface, p, KSO.PROJECTION)
+        self.optimization_utilities.AssignVectorToVariable(self.design_surface, c, KSO.CORRECTION)
+        self.optimization_utilities.AssignVectorToVariable(self.design_surface, p+c, KSO.SEARCH_DIRECTION)
+        self.optimization_utilities.AssignVectorToVariable(self.design_surface, p+c, KSO.CONTROL_POINT_UPDATE)
+        self.__LineSearch()
+
+
+
+    # --------------------------------------------------------------------------
+    def __getActiveConstraints(self):
+        active_constraint_values = []
+        active_constraint_variables = []
+        active_relaxation_coefficient = []
+        active_correction_coefficient = []
+
+        for constraint in self.constraints:
+            if self.__isConstraintActive(constraint):
+                identifier = constraint["identifier"].GetString()
+                constraint_value = self.communicator.getStandardizedValue(identifier)
+                buffer_value = self.constraint_buffer_variables[identifier]["buffer_value"]
+
+                active_constraint_values.append(constraint_value)
+                g_a_variable = self.constraint_gradient_variables[identifier]["mapped_gradient"]
+                g_a_norm = self.optimization_utilities.ComputeMaxNormOfNodalVariable(self.design_surface, g_a_variable)
+                g_a_variable_vector = KM.Vector()
+                self.optimization_utilities.AssembleVector(self.design_surface, g_a_variable_vector, g_a_variable)
+                if abs(g_a_norm) > 1e-10:
+                    g_a_variable_vector *= 1.0/g_a_norm
+
+                active_constraint_variables.append(g_a_variable_vector)
+                active_relaxation_coefficient.append(min(buffer_value,1.0))
+
+                max_buffer = 2.0
+                if buffer_value > 1.0:
+                    if buffer_value < max_buffer:
+                        active_correction_coefficient.append(2*(buffer_value - 1))
+                    else:
+                        active_correction_coefficient.append(2*(max_buffer-1))
+                else:
+                    active_correction_coefficient.append(0.0)
+
+
+        return active_constraint_values, active_constraint_variables, active_relaxation_coefficient, active_correction_coefficient
```

Comment:
```suggestion
        return active_constraint_values, active_constraint_variables, active_relaxation_coefficient, Kratos.Vector(active_correction_coefficient)
```

So you can avoid using `OptimizationUtilities::AssembleBufferVector` method. I would use `Krats.Vector(python::list)` everywhere rather than calling the method `OptimizationUtilities::AssembleBufferVector`